### PR TITLE
Enable ENGINEPRIME feature in deb packaging

### PR DESCRIPTION
**Pull request by mr-smidge**
Replaces use of git checkout with an HTTPS download, to work around errors in the deb/PPA build not finding git.

### CODE REVIEW

**Comment by mr-smidge** [comment]
To test, I did:

```
$ cpack -G External -D DEB_SOURCEPKG=ON
```

This change works fine on my normal development workstation (Linux Mint 20) and an Ubuntu 18.04 VM I tried, but I will happily take any advice on replicating the build server environment more accurately to ensure it doesn't break as soon as this PR is merged!

**Comment by daschuer** [comment]
Since this is the main branch I don't really mind if we break the PPA build for a short period.

**Review by daschuer — APPROVED** [review]
Let's give it a try. Thank you for taking care.

**Comment by Be-ing** [comment]
Let's give it a try. If this work, let's do the same for libkeyfinder.

**Comment by daschuer** [comment]
Can you remove the draft state or is there anything else to do?

**Comment by mr-smidge** [comment]
@Be-ing has already taken care of it :-).  Thanks both.

**Comment by daschuer** [comment]
It fails unfortunately: 
https://launchpadlibrarian.net/521866460/buildlog_ubuntu-groovy-amd64.mixxx_2.4.0~alpha~pre1~git7758-0ubuntu2~main~git7758~groovy_BUILDING.txt.gz

```
-- Using src='https://github.com/xsco/libdjinterop/archive/0.14.6.tar.gz'
CMake Error at libdjinterop-stamp/download-libdjinterop.cmake:159 (message):
  Each download failed!

    error: downloading 'https://github.com/xsco/libdjinterop/archive/0.14.6.tar.gz' failed
         status_code: 6
         status_string: "Couldn't resolve host name"
         log:
         --- LOG BEGIN ---
         Could not resolve host: github.com

  Closing connection 0
```

**Comment by Be-ing** [comment]
I guess the builder doesn't have any network access, quite reasonably. I think all the sources need to be in the package or provided by another package.

**Comment by daschuer** [comment]
So we have two options, provide libdjinterop as debian package or copy it to our lib folder. 
Both would work for me.

**Comment by uklotzde** [comment]
> So we have two options, provide libdjinterop as debian package or copy it to our lib folder.
> Both would work for me.

This is one of the reasons why packaging for specific distributions should better be separated from the source repo.

**Comment by Be-ing** [comment]
We should stop with this ugly hack of sticking libraries into our source tree. I am sure the Debian package maintainers would not like that. We need to package our dependencies.

**Comment by Be-ing** [comment]
The Debian maintainers will probably need to package libdjinterop anyway before updating Mixxx, so we should just do that now.

**Comment by mr-smidge** [comment]
Grovelling PR to revert: https://github.com/mixxxdj/mixxx/pull/3621

In the meantime, I'll get libdjinterop packaged for Debian: https://github.com/xsco/libdjinterop/issues/45

**Comment by uklotzde** [comment]
We could either host the libraries on our PPA in the first step or even include the code for static linking like we do for RPMFusion.

As long as no other applications depend on those libraries this is a valid choice. And we probably have very strict version requirements for libdjinterop that would make using a shared library difficult.

Please consider the fact that we are still supporting very old LTS releases with outdated dependencies. This only works if APIs are stable or at least backward compatible.

**Comment by uklotzde** [comment]
...thinking about to the future when we need to maintain our legacy dependencies, because we cannot upgrade the version that was once included in the distribution release.

**Comment by Holzhaus** [comment]
> We should stop with this ugly hack of sticking libraries into our source tree.

Completely agree.


> We need to package our dependencies.

Developers are not package maintainers. If you want to do the work, go ahead. But IMHO it's not our responsibility. And packaging on debian is a mess (one of the main reasons switched to Archlinux). 

Could we just use vcpkg for Linux dependencies as well?

**Comment by Be-ing** [comment]
> We could either host the libraries on our PPA in the first step or even include the code for static linking like we do for RPMFusion.

Yes I think we should put the package in our PPA and at the same time submit it upstream.

> Could we just use vcpkg for Linux dependencies as well?

We could, but mixing vcpkg and Debian packaging doesn't seem like a great idea. I think vcpkg could be a good idea for Flatpak though.

**Comment by Be-ing** [comment]
> Developers are not package maintainers. If you want to do the work, go ahead. But IMHO it's not our responsibility.

It is our responsibility as long as we have our own PPA.

**Comment by mr-smidge** [comment]
Are there any restrictions on how many simultaneous versions can be hosted on the PPA?

The reason for asking is that there is an open issue that will break the public API (libdjinterop is still in its 0.x phase) https://github.com/xsco/libdjinterop/issues/36, and such a change might require before and after versions to be available if a published Mixxx release depended on the earlier one.

**Comment by Holzhaus** [comment]
> > Developers are not package maintainers. If you want to do the work, go ahead. But IMHO it's not our responsibility.
> 
> It is our responsibility as long as we have our own PPA.

Yes. I sucks losing beta testers, but IMHO we should stop providing a PPA if it's such a burden. I have the impression we spent way more time with managing depenendencies then actually developing mixxx lately. Alas, I'm probably alone with that position.

**Comment by Be-ing** [comment]
> I have the impression we spent way more time with managing depenendencies then actually developing mixxx lately. Alas, I'm probably alone with that position.

No, that's an accurate description of what we've been doing lately. Until now this has been no big deal on Linux because we use libraries that are already packaged. Now we are adding two new dependencies that (so far) only we use.

**Comment by Be-ing** [comment]
If we discontinue the PPA and use Flatpak instead, we'll probably gain beta testers.

**Comment by daschuer** [comment]
> Are there any restrictions on how many simultaneous versions can be hosted on the PPA?

Yes, because you can have only one version on a PPA per distro version. However this does not hurt, because we have only one Mixxx version. It should just work together. Alternative, you can issue your own PPA and we can add it as external package source. 
That works as well.

**Pull request merged**
